### ISSUE TITLE
shortened modified names by certain characters in the package names

### ISSUE THREAD

**Issue by randomName0815** [BIG-RAT/jamfcpr#12]
hi, i have not been able to test the tool fully yet, because we get strange results when package names contain german characters like ä ö ü. are the names only displayed wrong in the list and can it still be used normally?
![image](https://user-images.githubusercontent.com/109347634/179200642-b148c3b9-6fb4-4a7f-a7fc-08768035795a.png)

btw. the app crashes when clicking into the empty list after launching the app (v3.3.3)

**Comment by BIG-RAT** [comment]
Thanks for the heads up.  Crash will be easy to fix.  Handling the special characters looks to be a bit more challenging. It seems the name returned from the API for the /packages endpoint differs from what is returned from /packages/id.  For example:
packages endpoint returns (expected results):
<img width="183" alt="image" src="https://user-images.githubusercontent.com/22251543/179420359-d8d2f388-d1f7-4496-a131-5f0969a84827.png">

packages/id/1945 yields:
<img width="329" alt="image" src="https://user-images.githubusercontent.com/22251543/179420417-8267df17-e51d-491d-aa8f-fb5a9b63d756.png">

Within the app log I see:
`[ViewController.fetchPackageList] Packages to replicate: ["̈öü.pkg"]`
It's like the a character is replaced with a backspace.

Note, if I copy from a locally stored file it looks ok.
```
[ViewController.fetchPackageList] Packages to replicate: ["äöü.pkg"]
[ViewController.fetchPackageList] sourcePackageRecord: ["checksum": "", "name": "äöü.pkg", "size": "14904", "hashType": "MD5"]
```


Can't say for sure if this is/relates to the issue yet.

**Comment by BIG-RAT** [comment]
My testing showed positive results for preserving the name/filename if it contains special characters. 
<img width="182" alt="image" src="https://user-images.githubusercontent.com/22251543/179463371-08c1a505-f78e-4e59-8583-1e4639be3b00.png">

 Crash should also be resolved in v3.3.4.
